### PR TITLE
Feature/client role permission services

### PR DESCRIPTION
**Pull request by Camilo-Camargo**
# Servicios de ClientRole, Role, RolePermission y Permission

## Description
Se agregaron los servicios para las siguientes entidades:

`ClientRole`:
- `Create`: crear asociación
- `Delete`: eliminar asociación
- `List`: listar IDs de la tabla asociativa
- `ListRolesByClientId`: listar roles por ID de cliente

`RolePermission`:
- `ListPermissionsByRoleId`: listar permisos por ID del rol
- `List`: listar IDs de asociaciones

`Role`:
- `List`: listar todos los roles

`Permission`:
- `List`: listar todos los permisos

### Summary
Se han desarrollado e implementado servicios para la gestión de roles y permisos, incluyendo la creación, eliminación y listado de asociaciones entre clientes, roles y permisos.

## How Has This Been Tested?
Se ha realizado una prueba manual.

### Test Configuration:
- **Docker**:*version 25.0.3, build 4debf41* 
- **Docker Compose**:*Docker Compose version v2.24.5-desktop.1* 
- **OS**:*MacOs*
- **Version**:*Sonoma 14.4.1*

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: internal/interfaces/role.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+	dto "optitech/internal/dto/roles"
+	models "optitech/internal/sqlc"
+)
+
+type IRoleService interface {
+	List() (*[]dto.GetRoleRes, error)
+}
+
+type IRoleRepository interface {
+	GetRole(id int64) (*dto.GetRoleRes, error)
```
Comment (CamiloArias12) [inline internal/interfaces/role.go:13]: No es mejor manejar int32 ?

File: internal/interfaces/role_permission.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	models "optitech/internal/sqlc"
+)
+
+type IRolePermissionService interface {
+	ListPermissionsByRoleId(roleId int32) (*[]p.GetPermissionRes, error)
+	List() (*[]dto.GetRolePermissionRes, error)
+}
+
+type IRolePermissionRepository interface {
+	GetRolePermission(id int64) (*dto.GetRolePermissionRes, error)
```
Comment (CamiloArias12) [inline internal/interfaces/role_permission.go:15]: No es mejor manejar int32 ?

Comment (Camilo-Camargo) [reply internal/interfaces/role_permission.go:15]: [a379c4c](https://github.com/OwlByTech/optitech-api/pull/18/commits/a379c4c0ba6394cb791eee72c55b9114a16ad0ff)

File: internal/interfaces/role.go (hunk 1 above)
Comment (Camilo-Camargo) [reply internal/interfaces/role.go]: [a379c4c](https://github.com/OwlByTech/optitech-api/pull/18/commits/a379c4c0ba6394cb791eee72c55b9114a16ad0ff)

File: internal/service/client_role.go
Comment (CamiloArias12) [inline internal/service/client_role.go:1]: los servicos deberian estar dentro de una carpeta para seguir la estructura ya definida

File: internal/interfaces/role_permission.go (hunk 2 above)
Comment (Camilo-Camargo) [reply internal/interfaces/role_permission.go:15]: rolepermission es la asociacion

File: internal/service/client_role.go
Comment (Camilo-Camargo) [reply internal/service/client_role.go:1]: [9dde96a](https://github.com/OwlByTech/optitech-api/pull/18/commits/9dde96af59ec02c24e0414a1003572bf64cdbb6b)

**Review by CamiloArias12 — APPROVED** [review]

**Review by JosephSC0121 — APPROVED** [review]

**Pull request merged**